Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -406,21 +411,21 @@ getInformation personId = do
       >>= fromMaybeM (MerchantNotFound merchantId.getId)
   pure $ makeDriverInformationRes driverEntity organization driverReferralCode
 
-setActivity :: (CacheFlow m r, EsqDBFlow m r) => Id SP.Person -> Bool -> m APISuccess.APISuccess
-setActivity personId isActive = do
+setActivity :: (CacheFlow m r, EsqDBFlow m r) => Id SP.Person -> Bool -> Maybe DriverInfo.DriverMode -> m APISuccess.APISuccess
+setActivity personId isActive mode = do
   _ <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
   let driverId = cast personId
-  when isActive $ do
+  when (isActive || (isJust mode && (mode == Just DriverInfo.SILENT || mode == Just DriverInfo.ONLINE))) $ do
     driverInfo <- QDriverInformation.findById driverId >>= fromMaybeM DriverInfoNotFound
     unless driverInfo.enabled $ throwError DriverAccountDisabled
     unless (not driverInfo.blocked) $ throwError DriverAccountBlocked
-  QDriverInformation.updateActivity driverId isActive
+  QDriverInformation.updateActivity driverId isActive mode
```

Comment:
can we keep the isActive as True and don't use it at all in pool calculation ? otherwise depreciating it will become difficult.